apiml-common/src/main/java/org/zowe/apiml/product/web/HttpConfig.java

```diff
@@ -289,12 +280,12 @@ public HostnameVerifier secureHostnameVerifier() {
 
     @Bean
     public EurekaJerseyClient eurekaJerseyClient() {
-        return eurekaJerseyClientBuilder.build();
+        return eurekaJerseyClientBuilder().get().build();
     }
 
     @Bean
-    public EurekaJerseyClientBuilder eurekaJerseyClientBuilder() {
-        return eurekaJerseyClientBuilder;
+    public Supplier<EurekaJerseyClientBuilder> eurekaJerseyClientBuilder() {
+        return () -> factory.createEurekaJerseyClientBuilder(eurekaServerUrl, serviceId);
```

Comment:
Why a supplier and not a straight up call to the factory?

Reply:
Because if there is just a builder the same instance is taken multiple times and always updated. We cannot be sure there is an expected configuration. This ensure the builder is always in the default state and nothing unexpected was set.